### PR TITLE
MOM6: Cleaned up vertical parameterization code

### PR DESCRIPTION
**Pull request by Hallberg-NOAA**
This PR includes a series of changes that address minor problems with the
vertical parameterization code or augment the dimensional consistency testing
within this code, including new options that correct bugs or update algorithms. 
By default, none of these updates change answers, but there are number of
changes to the MOM_parameter_doc files.  MOM6 commits with this PR include:

- NOAA-GFDL/MOM6@1b31d9d +Added SURFACE_FORCING_2018_ANSWERS
- NOAA-GFDL/MOM6@2578211 +Added WIND_GYRES_2018_ANSWERS
- NOAA-GFDL/MOM6@5cc85dc +Added DEFAULT_2018_ANSWERS
- NOAA-GFDL/MOM6@b470a27 Merge branch 'dev/gfdl' into parameterizations_cleanup
- NOAA-GFDL/MOM6@f258630 +Rescaled units of g_Earth in vertical params code
- NOAA-GFDL/MOM6@cca8046 Add g_Earth to all versions of surface_forcing_CS
- NOAA-GFDL/MOM6@48fc7f2 Rescaled units of fluxes%ustar
- NOAA-GFDL/MOM6@ac8b84a (*)Improved SW_trans correction for small fluxes
- NOAA-GFDL/MOM6@cbd6f99 Split excessively long lines in 2 files
- NOAA-GFDL/MOM6@a0d669d (*)Multiply fmax by US%s_to_T in MOM_hor_visc.F90
- NOAA-GFDL/MOM6@d36bdb8 Merge branch 'dev/gfdl' into parameterizations_cleanup
- NOAA-GFDL/MOM6@908cff6 +(*)Corrected documentation of BBL_MIXING_MAX_DECAY
- NOAA-GFDL/MOM6@a42140d +Added 3 new runtime parameters to MOM_opacity.F90
- NOAA-GFDL/MOM6@beaedaa +Added FRACTIONAL_ROUGHNESS_MAX run-time parameter
- NOAA-GFDL/MOM6@562e567 +Added nsw argument to applyBoundaryFluxesInOut
- NOAA-GFDL/MOM6@bf4c12c +Added optics extractor routines
- NOAA-GFDL/MOM6@d4b476e Explicitly declare the size of the pen_SW_bnd args
- NOAA-GFDL/MOM6@69a0d8a +Combined MOM_shortwave_abs and MOM_opacity
- NOAA-GFDL/MOM6@2640b93 +Set chlorophyll for shortwave in set_pen_shortwave
- NOAA-GFDL/MOM6@4937fe5 +Added the new subroutine set_pen_shortwave
- NOAA-GFDL/MOM6@bd201fc +Separate shortwave forcing to set_opacity
- NOAA-GFDL/MOM6@48adabd +Rescale timestep in shortwave heating routines
- NOAA-GFDL/MOM6@057dfde Corrected units in various comments
- NOAA-GFDL/MOM6@408f9cd +Changed bulkmixedlayer timestep arg units to T
- NOAA-GFDL/MOM6@be2da7b Refactored find_coupling_coef
- NOAA-GFDL/MOM6@2cedf63 +Added runtime parameter FRACTIONAL_ROUGHNESS_MAX
- NOAA-GFDL/MOM6@2e70663 Dimensional consistency in MOM_bulk_mixed_layer
- NOAA-GFDL/MOM6@4343b70 +Rescaled the units of Waves%KvS
- NOAA-GFDL/MOM6@f91b39c +Added runtime parameters for the bulk mixed layer
- NOAA-GFDL/MOM6@79e6902 +Added new runtime option TIDAL_MIXING_2018_ANSWERS
- NOAA-GFDL/MOM6@a132309 +Moved internal tide testing code
- NOAA-GFDL/MOM6@10d9f08 +Merged diabatic_ALE code into diabatic_ALE_legacy
- NOAA-GFDL/MOM6@2b36676 +Offer ePBL LT diagnostics only when EPBL_LT=True
- NOAA-GFDL/MOM6@84f02b7 Code rearrangement in MOM_diabatic_driver.F90
- NOAA-GFDL/MOM6@464b667 Split long comments in RGC_tracer.F90
- NOAA-GFDL/MOM6@cbe68e9 +Added the new subroutine diabatic_ALE_legacy
- NOAA-GFDL/MOM6@7931e98 +Eliminated the legacy_diabatic public interface
- NOAA-GFDL/MOM6@bc80b12 (*)Correct dimensional scaling with RESOLVE_EKMAN
- NOAA-GFDL/MOM6@1ca9998 (*)Properly rescale Tr_ea_BBL in legacy_diabatic
- NOAA-GFDL/MOM6@707442f (*)Removed hard-coded values in set_int_tide_input
- NOAA-GFDL/MOM6@ea25c1b +Altered arguments to Calculate_kappa_shear
- NOAA-GFDL/MOM6@4f224af Rescaled TKE in MOM_kappa_shear
- NOAA-GFDL/MOM6@be7a61f Rescaled frequencies in MOM_kappa_shear
- NOAA-GFDL/MOM6@fd4fb8b Rescaled internal variables in find_kappa_tke
- NOAA-GFDL/MOM6@61f04d5 Pass time step to kappa_shear_column in units of T
- NOAA-GFDL/MOM6@a618823 Partial dimensional testing in MOM_kappa_shear
- NOAA-GFDL/MOM6@7d9651c +Added dimensional testing for visc$ustar_BBL
- NOAA-GFDL/MOM6@949f622 Added dt_in_T to diabatic and legacy_diabatic
- NOAA-GFDL/MOM6@7ee45c4 +Added dimensional rescaling of visc%Ray_u
- NOAA-GFDL/MOM6@3834188 +Added dimensional testing in MOM_vert_friction.F90
- NOAA-GFDL/MOM6@ae1795d (*)Set I_2Omega with HENYEY_IGW_BACKGROUND_NEW
- NOAA-GFDL/MOM6@6f259d6 +Added dimensional testing in MOM_set_viscosity.F90
- NOAA-GFDL/MOM6@b2c6bab +Added dimensional testing for BBL viscosities
- NOAA-GFDL/MOM6@badeb50 +Added dimensional testing for shared viscosities
- NOAA-GFDL/MOM6@9184c36 +Yet more dimensional testing for diffusivities
- NOAA-GFDL/MOM6@e9cf2cd +Added dimensional testing for more diffusivities
- NOAA-GFDL/MOM6@5590d6b +Added dimensional testing for diffusivities
- NOAA-GFDL/MOM6@eb5f06c +Extended dimensional scaling of vertical params
- NOAA-GFDL/MOM6@6c9a442 +Added ML_RAD_BUG and SET_DIFF_2018_ANSWERS
- NOAA-GFDL/MOM6@9f14b08 Merge branch 'dev/gfdl' into parameterizations_cleanup
- NOAA-GFDL/MOM6@2288502 +Added new runtime option SET_VISC_2018_ANSWERS
- NOAA-GFDL/MOM6@93a7d94 +Add EPBL_VEL_SCALE_SCHEME & EPBL_VEL_SCALE_FACTOR
- NOAA-GFDL/MOM6@7c995db +Added EPBL_MSTAR_SCHEME and EPBL_LANGMUIR_SCHEME
- NOAA-GFDL/MOM6@e0d2671 +Change units of ustar in get_Langmuir_number
- NOAA-GFDL/MOM6@ef55110 Add dimensional consistency testing to ePBL_column
- NOAA-GFDL/MOM6@8f9d7e3 Call ePBL_column
- NOAA-GFDL/MOM6@798dec8 +Added ePBL_column, but it is not yet called.
- NOAA-GFDL/MOM6@ea6d355 Added an internal type in MOM_energetic_PBL
- NOAA-GFDL/MOM6@bcb6b88 +Removed 5 diagnostics from ePBL
- NOAA-GFDL/MOM6@853377e +(*)Eliminated ML_Depth2 from energetic_PBL_CS
- NOAA-GFDL/MOM6@75d2fd3 Reduced ePBL array dimensions
- NOAA-GFDL/MOM6@ab386bc +Added EPBL_2018_ANSWERS
- NOAA-GFDL/MOM6@63b2061 +Deleted unused code & options in MOM_energetic_PBL
- NOAA-GFDL/MOM6@f5c9aa0 Code clean-up in MOM_energetic_PBL
- NOAA-GFDL/MOM6@3eeabe2 Merge branch 'Reichl_ePBL_update' into parameterizations_cleanup
- NOAA-GFDL/MOM6@14fc76e Added parentheses to a sum in add_drag_diffusivity
- NOAA-GFDL/MOM6@4483c26 Updating comments in MOM_energetic_PBL.F90
- NOAA-GFDL/MOM6@7f9f75e Merge branch 'dev/gfdl' into user/bgr/OSBL_Updates_more
- NOAA-GFDL/MOM6@45dc3f6 Formatting/comment updates in MOM_energetic_PBL.F90
- NOAA-GFDL/MOM6@bf048da Reformatting in MOM_energetic_PBL
- NOAA-GFDL/MOM6@c652668 Replaced x**2.0 with x**2
- NOAA-GFDL/MOM6@9a3f3e2 Removed KDML and HMIX from MOM_set_diffusivity
- NOAA-GFDL/MOM6@a84c0e0 +Obsoleted USE_VISBECK_SLOPE_BUG
- NOAA-GFDL/MOM6@09a49a0 Unpacking Loops in ePBL for code clarity.
- NOAA-GFDL/MOM6@8d896f2 Rearrangement of ePBL, prior to pulling out ePBL_inner loop
- NOAA-GFDL/MOM6@8a2968b Merge branch 'user/bgr/OSBL_Updates' into user/bgr/OSBL_Updates_more
- NOAA-GFDL/MOM6@ea57802 First working version of split Mstar in MOM_ePBL

### CODE REVIEW

**Comment by Hallberg-NOAA** [comment]
This PR is being tested with https://gitlab.gfdl.noaa.gov/ogrp/MOM6/pipelines/8447

File: src/parameterizations/vertical/MOM_energetic_PBL.F90 [hunk 1]
```diff
@@ -1916,13 +1759,184 @@ subroutine find_PE_chg_orig(Kddt_h, h_k, b_den_1, dTe_term, dSe_term, &
...
+subroutine Mstar_Langmuir(CS, US, abs_Coriolis, buoyancy_flux, ustar, BLD, Langmuir_Number, &
+                          mstar, mstar_LT, Convect_Langmuir_Number)
+  type(energetic_PBL_CS), pointer    :: CS    !< Energetic_PBL control structure.
+  type(unit_scale_type), intent(in)  :: US    !< A dimensional unit scaling type
+  real,                  intent(in)  :: Abs_Coriolis !< Absolute value of the Coriolis parameter [T-1 ~> s-1]
+  real,                  intent(in)  :: Buoyancy_Flux !< Buoyancy flux [Z2 T-3 ~> m2 s-3]
+  real,                  intent(in)  :: UStar !< Surface friction velocity with? gustiness [Z T-1 ~> m s-1]
+  real,                  intent(in)  :: BLD   !< boundary layer depth [Z ~> m]
+  real,                  intent(inout) :: Mstar !< Input/output mstar (Mixing/ustar**3) [nondim]
+  real,                  intent(in)  :: Langmuir_Number !Langmuir number [nondim]
```
Comment (adcroft) [inline src/parameterizations/vertical/MOM_energetic_PBL.F90]: This line has incorrect syntax for doxygen markup: the "!" should be "!< " (including the white space please).

**Comment by Hallberg-NOAA** [comment]
The requested change has been made, and the revised version is being tested at https://gitlab.gfdl.noaa.gov/ogrp/MOM6/pipelines/8450.

File: src/parameterizations/vertical/MOM_energetic_PBL.F90 (hunk 1 above)
Comment (Hallberg-NOAA) [reply src/parameterizations/vertical/MOM_energetic_PBL.F90]: The requested modification has been changed, and the revised pull request is now passing all checks.

**Pull request merged**